### PR TITLE
Document gopass extension model

### LINKED ISSUE #2290: Document gopass extension framework for custom commands
This line in the README implies that custom commands should be possible via extensions.

https://github.com/gopasspw/gopass/blob/d6acc7063de4e7ceee717284dd1e4e3009163b5b/README.md?plain=1#L47

However, there is no further documentation either for creating extensions nor registering custom commands anywhere in the repository.

### PR DESCRIPTION
**Pull request by dominikschulz**
Fixes #2290

RELEASE_NOTES=[ENHANCEMENT] Document extension model.

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>

### CODE REVIEW

File: docs/hacking.md [hunk 1]
```diff
@@ -37,3 +37,9 @@ $ export GOPASS_HOMEDIR=/tmp/gp1
 $ mkdir -p $GOPASS_HOMEDIR
 $ go build && ./gopass setup --crypto age --storage gitfs
 ```
+
+## Extending gopass
+
+The main extension model small binaries that use the [gopass API](https://pkg.go.dev/github.com/gopasspw/gopass/pkg/gopass/api) package. This package provides a small and easy to use API that should work with any up to date gopass setup.
```
Comment (AnomalRoil) [inline docs/hacking.md:43]: > The main extension model small binaries 

I find this difficult to understand, is it missing a few parts?

File: docs/hacking.md [hunk 2]
```diff
@@ -37,3 +37,9 @@ $ export GOPASS_HOMEDIR=/tmp/gp1
...
+
+We don't have extensive documentation for this, yet. But the [gopass-hibp](https://github.com/gopasspw/gopass-hibp/blob/master/main.go) binary should provide an easy example that can be used as a blueprint.
```
Comment (AnomalRoil) [inline docs/hacking.md:45]: Looking at this example, I don't really understand how to "connect" to my Gopass neither. 
Is it just running Gopass as is under the hood, using the same Env variable and same config and config folder as the Gopass binary?

We might want to precise how this "works together" somewhere I think.

**Review by AnomalRoil — DISMISSED** [review]
Looking at https://pkg.go.dev/github.com/gopasspw/gopass/pkg/gopass/api, I feel like I don't understand how I'm supposed to be "connecting" to a Gopass instance using this API. See my last review comment too.

Otherwise, LGTM.

File: docs/hacking.md (hunk 1 above)
Comment (dominikschulz) [reply docs/hacking.md]: Yeah, I guess so.

File: docs/hacking.md (hunk 2 above)
Comment (dominikschulz) [reply docs/hacking.md]: Should be better now.

**Pull request merged**